### DISCUSSION TITLE
Crop controls always visible and allow to move an element that is behind another v6.

### DISCUSSION THREAD

**Discussion by marcos-velasquez** [fabricjs/fabric.js#9107] [General]
I made a custom object that extends from fabric Image to give it the ability to clip.

I have two problems that I cannot solve and I need your help.

1) How do I keep custom controls always visible.

2) How can I move the image (clone) that is behind the original image (movement `locked)`

what i currently have [https://github.com/marcos-velasquez/fabricjs-crop](https://github.com/marcos-velasquez/fabricjs-crop)

![Captura de pantalla 2023-07-18 225948](https://github.com/fabricjs/fabric.js/assets/39884122/0b0aa5e4-a3bb-491a-bdf9-9d2770f2c120)
![Captura de pantalla 2023-07-18 230011](https://github.com/fabricjs/fabric.js/assets/39884122/a77a98f3-c994-4b14-80ab-8ade6d256d5d)

An example of how I want it to work [shutterstock](https://www.shutterstock.com/)

![Captura de pantalla 2023-07-18 225447](https://github.com/fabricjs/fabric.js/assets/39884122/f5580fe4-3924-4899-9cb2-880d084b66a0)

Thank you for your help and for this excellent library.

**Comment by ShaMan123** [comment]
> 1. How do I keep custom controls always visible.

Override `Canvas#drawControls`



> 2\. How can I move the image (clone) that is behind the original image (movement `locked)`

@asturur can answer that
If I am not mistaken they use a single object so all the controls you see belong to it - it is simpler.

**Reply by marcos-velasquez** [reply]
The first thing solved thank you very much.

```
import { canvas } from '@fabric-sdk';
import { Canvas, Object } from 'fabric';

const defaultFunction = Canvas.prototype.drawControls;

export class ObjectControl {
  constructor(private object: Object) {}

  public static restoreDefaultBehavior() {
    Canvas.prototype.drawControls = defaultFunction;
  }

  public showControlAlways() {
    Canvas.prototype.drawControls = (ctx: CanvasRenderingContext2D) => {
      const activeObject = canvas().getActiveObject();

      if (activeObject) {
        activeObject._renderControls(ctx);
      }

      if (activeObject !== this.object) {
        this.object._renderControls(ctx);
      }
    };
  }
}
```

The second thing I have to investigate. Any help is greatly appreciated.
The developed crop logic is handled with the original image (without movement) and a clone. In this case, there is no alternative to move the clone when trying to move the image that is above it?


Thank you so much.
